### ISSUE TITLE
JACK detection is not working properly

### ISSUE THREAD

**Issue by barni2000** [msm8953-mainline/linux#197]
The mhbc thresholds was updated per device to match downstream because it was flaky, but unfortunately it not working correctly. On the most device in evtest only headphones state changing and a headset mic is undetected.

This was the related changes:
https://github.com/msm8953-mainline/linux/compare/3e4e830c8f21497d5a69f66be87491340a5ce3c2..26feecd5e199f347c5f713eab4d536662d7df7ef

**Comment by llg179** [comment]
It has been fixed for Fairphone 3 as well with full slimbus audio (headset, mics; call-voice is in fp3-7.0.9-voice branch). 
https://github.com/msm8953-mainline/linux/compare/7.0.9/main...llg179:linux:wcd9335-mbhc
Is it ok to upstream with AI assisted work? PmOS policy prohibits that. 
https://docs.postmarketos.org/policies-and-processes/development/ai-policy.html

**Comment by barni2000** [comment]
> It has been fixed for Fairphone 3 as well with full slimbus audio (headset, mics; call-voice is in fp3-7.0.9-voice branch). 
> https://github.com/msm8953-mainline/linux/compare/7.0.9/main...llg179:linux:wcd9335-mbhc
> Is it ok to upstream with AI assisted work? PmOS policy prohibits that. 
> https://docs.postmarketos.org/policies-and-processes/development/ai-policy.html

FP3 is using different audio architecture and we don't merge AI assisted work, it is only allowed at upstream.

**Comment by llg179** [comment]
> FP3 is using different audio architecture and we don't merge AI assisted work, it is only allowed at upstream.

Do you have the capacity to review the work and sign-off the commits to mainline mentioning me as co-author in the commits and Claude "Assisted-by: Claude:claude-opus-4.8"?

**Comment by barni2000** [comment]
> > FP3 is using different audio architecture and we don't merge AI assisted work, it is only allowed at upstream.
> 
> Do you have the capacity to review the work and sign-off the commits to mainline mentioning me as co-author in the commits and Claude "Assisted-by: Claude:claude-opus-4.8"?

No, https://mastodon.social/@barni2000/116161573857090786

**Comment by llg179** [comment]
You're right, and thanks for the correction. I conflated two different things.

  This issue is about the MBHC in `sound/soc/codecs/msm8916-wcd-analog.c`, i.e. the
  pm8953 analog codec on MI2S, which is what every other msm8953/sdm632 device in
  the tree uses and what the per-device `qcom,mbhc-vthreshold-*` overrides tune.
  The FP3 is the only device here with a WCD9335 on SLIMbus, so my change is in
  `wcd9335.c` - different driver, different register map, different bus. It does
  not fix this issue and cannot affect it. Sorry for the noise in this thread.

  One thing that may or may not be useful for #197: on wcd9335 I had to rewrite
  `MECH_DETECT_TYPE` on *every* edge to re-arm L_DET, otherwise removal never
  fires an IRQ again. `msm8916-wcd-analog.c` already handles that via
  `MECH_DET_TYPE_INSERTION`/`REMOVAL`, so it's probably not the cause here -
  and given the symptom is that the mechanical state is seen but the headset mic
  is not, the electrical path (`MBHC_MIC_ELECTRICAL_INS_REM_DET`) plus the
  vthreshold calibration look like the more likely place to dig. That's a guess
  from a different driver, not something I've reproduced, so treat it as such.

  On the AI question: understood, and I won't send AI-assisted work here. I'll
  take the wcd9335 part to the ASoC list instead, disclosed per
  `Documentation/process/generated-content.rst` with an `Assisted-by:` tag. For
  what it's worth, `wcd9335.c` currently has no jack registration at all (no
  `snd_soc_jack`/`set_jack` anywhere in the driver), so this is a gap that also
  affects apq8096-db820c, the OnePlus 3/3T and the msm8996 Xiaomi devices — not
  just the FP3.
